### ISSUE TITLE
[Question] Inquiry about estimation of training memory size for mixed precision training 

### ISSUE THREAD

**Issue by callanwu** [huggingface/accelerate#2659]
Hello Accelerate team,

I am exploring the memory estimation logic for mixed precision training implemented in [estimate.py](https://github.com/huggingface/accelerate/blob/main/src/accelerate/commands/estimate.py#L215-L247)
and I have a question regarding the assignment of memory sizes.

In the code, the memory sizes are set as follows:

memory_sizes["model"] = fp32_size
memory_sizes["gradients"] = fp32_size + fp16_size
memory_sizes["optimizer"] = fp32_size * 2

However, according to the paper "ZeRO-Offload: Democratizing Billion-Scale Model Training", the memory requirements for different types of data (parameters, gradients, momentums, variances) in a mixed precision setting are different:

parameter16
gradient16
parameter32
momentum32
variance32

Based on this paper figure 2, if model parameters are assumed to be $\alpha$ GB, then the memory requirement should be 16* $\alpha$ (2+2+4+4+4=16)GB.

However, in the paper "ZeRO-Infinity: Breaking the GPU Memory Wall for Extreme Scale Deep Learning", section 3 specifies that each parameter requires 20 bytes of memory.

I am quite confused about this.

Could you please clarify the calculation of each memory_sizes entry in your code?
Thank you for your time and assistance.

https://github.com/huggingface/accelerate/blob/c0a37015e3e4f2ab78a35efd6462e5b5140cf46e/src/accelerate/commands/estimate.py#L215-L247

Reference:
[ZeRO-Offload: Democratizing Billion-Scale Model Training](https://arxiv.org/pdf/2101.06840.pdf)
[ZeRO-Infinity: Breaking the GPU Memory Wall for Extreme Scale Deep Learning](https://arxiv.org/pdf/2104.07857.pdf)
https://github.com/microsoft/DeepSpeed/issues/1137

## Who can help?
@muellerzr 
@SunMarc

**Comment by callanwu** [comment]
Need to be addressed.

**Comment by bbruceyuan** [comment]
hi, this confued me too. And do you figure it out now?   @callanwu

**Comment by callanwu** [comment]
> hi, this confued me too. And do you figure it out now? @callanwu

@bbruceyuan Hi, I think it depends on the type of gradient used by the optimizer, such as torch.float32, torch.float16, and torch.bfloat16.
Perhaps you can refer to [the official PyTorch documentation](https://pytorch.org/docs/stable/_modules/torch/optim/optimizer.html#Optimizer) on _fused_doc for more details.
